sql/core/src/main/scala/org/apache/spark/sql/execution/Columnar.scala

```diff
@@ -542,13 +524,17 @@ case class ApplyColumnarRulesAndInsertTransitions(
   /**
    * Inserts RowToColumnarExecs and ColumnarToRowExecs where needed.
    */
-  private def insertTransitions(plan: SparkPlan): SparkPlan = {
-    if (plan.supportsColumnar) {
+  private def insertTransitions(plan: SparkPlan, outputsColumnar: Boolean): SparkPlan = {
+    if (outputsColumnar) {
+      // The tree feels kind of backwards
+      // This is the end of the columnar processing so go back to rows
```

Comment:
This comment is incorrect. We are not going back to rows here.

Reply:
oh, I'll remove this comment